test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2741,6 +2741,52 @@ def test_cgroup_enforce_memsw(self):
         self.logger.info('Joined stdout/stderr contained expected string: '
                          + foundstr)
 
+    def test_cgroup_diag_messages(self):
+        """
+        Test to verify that job that exceeded resources has the diag_message
+        set correctly.
+        """
+
+        if not self.paths[self.hosts_list[0]]['memory']:
+            self.skipTest('Test requires memory subystem mounted')
+        # run the test if swap space is available
+        if not self.mem or not self.swapctl:
+            self.skipTest('Test requires memory controller with memsw'
+                          'swap accounting enabled')
+        if have_swap() == 0:
+            self.skipTest('no swap space available on the local host')
+        # Get the grandparent directory
+        fn = self.paths[self.hosts_list[0]]['memory']
+        fn = os.path.join(fn, 'memory.memsw.limit_in_bytes')
+        if not self.is_file(fn, self.hosts_list[0]):
+            self.skipTest('vmem resource not present on node')
+
+        # Make sure job history is enabled to see when job is gone
+        a = {'job_history_enable': 'True'}
+        rc = self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.assertEqual(rc, 0)
+        self.server.expect(SERVER, {'job_history_enable': 'True'})
+
+        self.load_config(self.cfg3 % ('', 'false', '', self.mem, '',
+                                      self.swapctl, ''))
+
+        a = {
+            'Resource_List.select':
+            '1:ncpus=1:mem=400mb:vmem=420mb:host=%s' % self.hosts_list[0]}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.eatmem_job1)
+        jid = self.server.submit(j)
+        a = {'job_state': 'F'}
+        self.server.expect(JOB, a, jid, extend='x')
+        resc = ['resources_used.diag_messages']
+        s = self.server.status(JOB, resc, id=jid, extend='x')
+        dmsg = s[0]['resources_used.diag_messages'].replace("'", "")
+        self.logger.info(dmsg)
```

Comment:
How about f"diag_message: {dmsg}"
Just printing a variable out is confusing.
If this is a dev debug message, consider removing it.

Reply:
You are right, it was a debug msg. Thank you for pointing out. Removed.